[DOCS] 8.4.2 Release notes (#2455)
elastic/security-docs: docs/release-notes/8.4.asciidoc

[[release-notes-header-8.4.0]]
== 8.4

[discrete]
[[release-notes-8.4.2]]
=== 8.4.2

[discrete]
[[bug-fixes-8.4.2]]
==== Bug fixes and enhancements

> * Removes access to the **Notes** and **Pinned** tabs in Timeline templates ({pull}140478[#140478]).

Reviewer: Do we need to update any other docs, if this removes functionality in the Timeline templates page?

Author: Good catch. Just a couple screenshots in this topic: https://www.elastic.co/guide/en/security/master/timeline-templates-ui.html#create-timeline-template

Author: Addressing this in https://github.com/elastic/security-docs/issues/2467.

* Fixes a bug with the **Attach to existing case** option in Timeline ({pull}139929[#139929]).
* Fixes bugs in the Rules table that affected the selected rule count and bulk select feature ({pull}139461[#139461]).
* Fixes a bug that caused the Rules page to incorrectly notify users that a prebuilt rules update was available, even after rules were fully updated ({pull}139287[#139287]).

> * Fixes a bug that prevented users from accessing alert details if they didn't have the appropriate privileges to view the internal index `.internal.alerts-security.alerts-spaceId`. Now, the Alert details flyout correctly uses the public alias index `.alerts-security,akerts-spaceId` ({pull}138331[#138331]).

Reviewer: Couple of typos -- I guessed on the comma/period based on the linked PR might want to confirm with dev:
```suggestion
* Fixes a bug that prevented users from accessing alert details if they didn't have the appropriate privileges to view the internal index `.internal.alerts-security.alerts-spaceId`. Now, the Alert details flyout correctly uses the public alias index `.alerts-security.alerts-spaceId` ({pull}138331[#138331]).
```

Author: Pretty sure your corrections are right. @michaelolo24 would you mind confirming?

Reviewer 2: Yep, it looks good. Thanks!

Author: We typically use `<Kibana-space>` , for example: https://www.elastic.co/guide/en/security/master/security-spaces.html

[[release-notes-header-8.4.0]]
== 8.4

[discrete]
[[release-notes-8.4.2]]
=== 8.4.2

[discrete]
[[known-issue-8.4.2]]
==== Known issues

> * A new Lucene 9 validation change may cause EQL rule errors whenever rule queries contain regular expressions using wildcard fields and predefined character classes (for example, `\w`, `\s`, `\d`).

Reviewer 3: Technically the rule is called event correlation, but maybe we can put `"may cause event correlation (EQL) rule errors...`?
```suggestion
* A new Lucene 9 validation change may cause event correlation rule errors whenever rule queries contain regular expressions using wildcard fields and predefined character classes (for example, `\w`, `\s`, `\d`).
```

Author: Yeah, will do. Good suggestion, thanks!

[discrete]
[[bug-fixes-8.4.2]]
==== Bug fixes and enhancements
* Removes access to the **Notes** and **Pinned** tabs in Timeline templates ({pull}140478[#140478]).
* Fixes a bug with the **Attach to existing case** option in Timeline ({pull}139929[#139929]).
* Fixes bugs in the Rules table that affected the selected rule count and bulk select feature ({pull}139461[#139461]).
* Fixes a bug that caused the Rules page to incorrectly notify users that a prebuilt rules update was available, even after rules were fully updated ({pull}139287[#139287]).
* Fixes a bug that prevented users from accessing alert details if they didn't have the appropriate privileges to view the internal index `.internal.alerts-security.alerts-spaceId`. Now, the Alert details flyout correctly uses the public alias index `.alerts-security,akerts-spaceId` ({pull}138331[#138331]).

> * Aligns warning message title with the warning icon ({pull}140719[#140719]).

Reviewer 3: ```suggestion
* Aligns the warning message title on the Rule details page with the warning icon ({pull}140719[#140719]).
```
